Audio-only m3u8 with alt-audio missing CODECS

### What version of Hls.js are you using?

1.6.0

### What browser (including version) are you using?

Chrome Version 140.0.7339.81 (Official Build) (64-bit)

### What OS (including version) are you using?

Windows 11

### Test stream

https://0fdb852a.hls-js-dev.pages.dev/demo/?src=%20https%3A%2F%2Fcdnapisec.kaltura.com%2Fp%2F4393802%2Fsp%2F439380200%2FplayManifest%2FentryId%2F1_571hu7y1%2Fprotocol%2Fhttps%2Fformat%2Fapplehttp%2FflavorIds%2F1_f63xps6e%2C1_rtgks9uv%2C1_h1y1q1p9%2Fa.m3u8%3FuiConfId%3D57209222%26playSessionId%3D71db95e8-3847-9d5c-201e-c223fd61d8cf%3A7d3d79b8-f618-47a9-706f-60a039591df6%26referrer%3D%26clientTag%3Dhtml5%3Av7.192&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsInN0b3BPblN0YWxsIjpmYWxzZSwiZHVtcGZNUDQiOmZhbHNlLCJsZXZlbENhcHBpbmciOi0xLCJsaW1pdE1ldHJpY3MiOi0xfQ==

### Configuration

```JavaScript
{
  "debug": true,
  "enableWorker": true,
  "lowLatencyMode": true,
  "backBufferLength": 90
}
```

### Additional player setup steps

_No response_

### Checklist

- [x] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [x] The issue occurs in the stable client (latest release) on https://hlsjs.video-dev.org/demo and not just on my page
- [x] The issue occurs in the latest client (main branch) on https://hlsjs-dev.video-dev.org/demo and not just on my page
- [x] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [x] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

play the manifest [1.6.0](https://0fdb852a.hls-js-dev.pages.dev/demo/?src=%20https%3A%2F%2Fcdnapisec.kaltura.com%2Fp%2F4393802%2Fsp%2F439380200%2FplayManifest%2FentryId%2F1_571hu7y1%2Fprotocol%2Fhttps%2Fformat%2Fapplehttp%2FflavorIds%2F1_f63xps6e%2C1_rtgks9uv%2C1_h1y1q1p9%2Fa.m3u8%3FuiConfId%3D57209222%26playSessionId%3D71db95e8-3847-9d5c-201e-c223fd61d8cf%3A7d3d79b8-f618-47a9-706f-60a039591df6%26referrer%3D%26clientTag%3Dhtml5%3Av7.192&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsInN0b3BPblN0YWxsIjpmYWxzZSwiZHVtcGZNUDQiOmZhbHNlLCJsZXZlbENhcHBpbmciOi0xLCJsaW1pdE1ldHJpY3MiOi0xfQ==) 

This is working on [1.5.8 -demo test](https://e867d9c3.hls-js-dev.pages.dev/demo/?src=%20https%3A%2F%2Fcdnapisec.kaltura.com%2Fp%2F4393802%2Fsp%2F439380200%2FplayManifest%2FentryId%2F1_571hu7y1%2Fprotocol%2Fhttps%2Fformat%2Fapplehttp%2FflavorIds%2F1_f63xps6e%2C1_rtgks9uv%2C1_h1y1q1p9%2Fa.m3u8%3FuiConfId%3D57209222%26playSessionId%3D71db95e8-3847-9d5c-201e-c223fd61d8cf%3A7d3d79b8-f618-47a9-706f-60a039591df6%26referrer%3D%26clientTag%3Dhtml5%3Av7.192&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsInN0b3BPblN0YWxsIjpmYWxzZSwiZHVtcGZNUDQiOmZhbHNlLCJsZXZlbENhcHBpbmciOi0xLCJsaW1pdE1ldHJpY3MiOi0xfQ==)

Issue is also reproducible on the latest version - [latest](https://hlsjs.video-dev.org/demo/?src=%20https%3A%2F%2Fcdnapisec.kaltura.com%2Fp%2F4393802%2Fsp%2F439380200%2FplayManifest%2FentryId%2F1_571hu7y1%2Fprotocol%2Fhttps%2Fformat%2Fapplehttp%2FflavorIds%2F1_f63xps6e%2C1_rtgks9uv%2C1_h1y1q1p9%2Fa.m3u8%3FuiConfId%3D57209222%26playSessionId%3D71db95e8-3847-9d5c-201e-c223fd61d8cf%3A7d3d79b8-f618-47a9-706f-60a039591df6%26referrer%3D%26clientTag%3Dhtml5%3Av7.192&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsInN0b3BPblN0YWxsIjpmYWxzZSwiZHVtcGZNUDQiOmZhbHNlLCJsZXZlbENhcHBpbmciOi0xLCJsaW1pdE1ldHJpY3MiOi0xfQ==)

### Expected behaviour

m3u8 should play

### What actually happened?

it is not playing, we only see a spinner 

### Console output

```shell
[log] > destroy
hls.ts:478 [log] > detachMedia
buffer-controller.ts:462 [log] > [buffer-controller]: media source detaching
base-stream-controller.ts:2093 [log] > [stream-controller]: IDLE->STOPPED
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: PARSED->STOPPED
blob:https://0fdb852a.hls-js-dev.pages.dev/7b54a602-13ab-44e8-aa46-44bca3cf7ac9:1076 [log] > Debug logs enabled for "main" in hls.js version 1.6.0
blob:https://0fdb852a.hls-js-dev.pages.dev/7b54a602-13ab-44e8-aa46-44bca3cf7ac9:1076 [log] > Debug logs enabled for "audio" in hls.js version 1.6.0
base-stream-controller.ts:2093 [log] > [subtitle-stream-controller]: IDLE->STOPPED
main.js:350 Using Hls.js config: {debug: true, enableWorker: true, lowLatencyMode: true, backBufferLength: 90}
logger.ts:102 [log] > Debug logs enabled for "Hls instance" in hls.js version 1.6.0
hls.ts:575 [log] > stopLoad
hls.ts:509 [log] > loadSource:https://cdnapisec.kaltura.com/p/4393802/sp/439380200/playManifest/entryId/1_571hu7y1/protocol/https/format/applehttp/flavorIds/1_f63xps6e,1_rtgks9uv,1_h1y1q1p9/a.m3u8?uiConfId=57209222&playSessionId=71db95e8-3847-9d5c-201e-c223fd61d8cf:7d3d79b8-f618-47a9-706f-60a039591df6&referrer=&clientTag=html5:v7.192
stream-controller.ts:611 [log] > [stream-controller]: Trigger BUFFER_RESET
hls.ts:462 [log] > attachMedia
buffer-controller.ts:320 [log] > [buffer-controller]: created media source: MediaSource
buffer-controller.ts:1508 [log] > [buffer-controller]: Media source opened
buffer-controller.ts:1337 [log] > [buffer-controller]: checkPendingTracks (pending: 0 codec events expected: 0) {}
level-controller.ts:370 [log] > [level-controller]: manifest loaded, 2 level(s) found, first bitrate: 68006
buffer-controller.ts:266 [log] > [buffer-controller]: 2 bufferCodec event(s) expected.
abr-controller.ts:798 [log] > [abr]: picked start tier {"videoRanges":[],"preferHDR":false,"minFramerate":0,"minBitrate":500000,"minIndex":null}
abr-controller.ts:958 [info] > [abr]: switch candidate:0->1 adjustedbw(500000)-bitrate=365033 ttfb:0.1 avgDuration:0.0 maxFetchDuration:4.0 fetchDuration:0.2 firstSelection:true codecSet: videoRange:SDR hls.loadLevel:-1
playlist-loader.ts:405 [log] > auto startLoad with configured startPosition -1
hls.ts:552 [log] > startLoad(-1)
level-controller.ts:463 [log] > [level-controller]: Switching to level 1 (SDR @134967) from level -1
audio-track-controller.ts:187 [log] > [audio-track-controller]: Updating audio tracks, 1 track(s) found in group(s): audio
audio-track-controller.ts:326 [log] > [audio-track-controller]: Switching to audio-track 0 "English" lang:en group:audio channels:undefined
base-stream-controller.ts:1919 [log] > [audio-stream-controller]: Reset loading state
level-controller.ts:668 [log] > [level-controller]: Loading level index 1 https://cfvod.kaltura.com/scf/hlsr/p/4393802/sp/439380200/serveFlavor/entryId/1_571hu7y1/v/1/ev/2/flavorId/1_h1y1q1p9/name/a.mp4/index.m3u8?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZnZvZC5rYWx0dXJhLmNvbS9zY2YvaGxzci9wLzQzOTM4MDIvc3AvNDM5MzgwMjAwL3NlcnZlRmxhdm9yL2VudHJ5SWQvMV81NzFodTd5MS92LzEvZXYvMioiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE3NTcwNTY4OTZ9fX1dfQ__&Signature=FzbSbU55LyiJKLi82JUCFR9Qf6ZKw75YqQc5oT3q~B~LI~VYd1qbJTEnE58ycTUDRQF62ijyK2o6~zNbqYwOtKO5ZmCPXswnxUOipTWE~qCdymyuF2OWA5dioyKTzRZJM1pu85u2ADItvPIFXYXy~Hi0nweNX8XlC8TjctPx1JRbccgSyYKaI375CTGKvE9o64TVROB44dV05Rm8zrazHXR8Ya4ki6ptmNDCTX~Q5k0t6SJP7Rr-0Udbgae0ZZZDxG62V8BxaQHNhH7m6cI0ubNtrl5sfzC3p0oKpQKfpZvfFogb0wZmn0LmFnuZehkePCMviuy6pqXPjOeEP6UvyQ__&Key-Pair-Id=APKAJT6QIWSKVYK3V34A
base-stream-controller.ts:2093 [log] > [stream-controller]: STOPPED->IDLE
audio-track-controller.ts:421 [log] > [audio-track-controller]: Loading audio-track 0 "English" lang:en group:audio https://cfvod.kaltura.com/scf/hlsr/p/4393802/sp/439380200/serveFlavor/entryId/1_571hu7y1/v/1/ev/2/flavorId/1_rtgks9uv/name/a.mp4/index.m3u8?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZnZvZC5rYWx0dXJhLmNvbS9zY2YvaGxzci9wLzQzOTM4MDIvc3AvNDM5MzgwMjAwL3NlcnZlRmxhdm9yL2VudHJ5SWQvMV81NzFodTd5MS92LzEvZXYvMioiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE3NTcwNTY4OTZ9fX1dfQ__&Signature=FzbSbU55LyiJKLi82JUCFR9Qf6ZKw75YqQc5oT3q~B~LI~VYd1qbJTEnE58ycTUDRQF62ijyK2o6~zNbqYwOtKO5ZmCPXswnxUOipTWE~qCdymyuF2OWA5dioyKTzRZJM1pu85u2ADItvPIFXYXy~Hi0nweNX8XlC8TjctPx1JRbccgSyYKaI375CTGKvE9o64TVROB44dV05Rm8zrazHXR8Ya4ki6ptmNDCTX~Q5k0t6SJP7Rr-0Udbgae0ZZZDxG62V8BxaQHNhH7m6cI0ubNtrl5sfzC3p0oKpQKfpZvfFogb0wZmn0LmFnuZehkePCMviuy6pqXPjOeEP6UvyQ__&Key-Pair-Id=APKAJT6QIWSKVYK3V34A
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: STOPPED->WAITING_TRACK
base-stream-controller.ts:2093 [log] > [subtitle-stream-controller]: STOPPED->IDLE
stream-controller.ts:671 [log] > [stream-controller]: Level 1 loaded [1,9][part-9--1], cc [0, 0] duration:58.03
base-stream-controller.ts:1725 [log] > [stream-controller]: setting startPosition to 0 by default
interstitials-controller.ts:1056 [log] > [interstitials]: setSchedulePosition 0, undefined
interstitials-controller.ts:1842 [log] > [interstitials]: buffered to boundary [primary: 0.00-58.03]
interstitials-controller.ts:1294 [log] > [interstitials]: resuming [primary: 0.00-58.03]
buffer-controller.ts:1314 [log] > [buffer-controller]: Updating MediaSource duration to 58.030
base-stream-controller.ts:899 [log] > [stream-controller]: Loading main sn: 1 of level 1 (frag:[0.000-2.000]) cc: 0 [1-9], target: 0
base-stream-controller.ts:2093 [log] > [stream-controller]: IDLE->FRAG_LOADING
audio-track-controller.ts:101 [log] > [audio-track-controller]: Audio track 0 "English" lang:en group:audio loaded [1-9]
audio-stream-controller.ts:550 [log] > [audio-stream-controller]: Audio track 0 "English" of "audio" loaded [1,9][part-9--1],duration:58.03
base-stream-controller.ts:1725 [log] > [audio-stream-controller]: setting startPosition to 0 by default
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: WAITING_TRACK->IDLE
base-stream-controller.ts:899 [log] > [audio-stream-controller]: Loading audio sn: 1 of track 0 (frag:[0.000-2.000]) cc: 0 [1-9], target: 0
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:88 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for main sn: 1 level: 1 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 0
        initSegmentChange: true
base-stream-controller.ts:512 [log] > [stream-controller]: Loaded main sn: 1 of level 1
transmuxer-interface.ts:88 [log] > injecting Web Worker for "audio"
audio-stream-controller.ts:677 [log] > [audio-stream-controller]: Unknown video PTS for cc 0, waiting for video PTS before demuxing audio frag 1 of [1 ,9],track 0
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: FRAG_LOADING->WAITING_INIT_PTS
base-stream-controller.ts:512 [log] > [audio-stream-controller]: Loaded audio sn: 1 of track 0
91827bb7-6e3c-4997-8e63-86119685b329:1076 [log] > Debug logs enabled for "main" in hls.js version 1.6.0
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
91827bb7-6e3c-4997-8e63-86119685b329:10721 [log] > manifest codec:undefined, parsed codec:mp4a.40.2, channels:2, rate:44100 (ADTS object type:2 sampling index:4)
91827bb7-6e3c-4997-8e63-86119685b329:1076 [log] > Debug logs enabled for "audio" in hls.js version 1.6.0
base-stream-controller.ts:2093 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1423 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2//mp4a.40.2]
buffer-controller.ts:590 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 0)
buffer-controller.ts:1337 [log] > [buffer-controller]: checkPendingTracks (pending: 1 codec events expected: 2) {"audio":{"listeners":[],"codec":"mp4a.40.2","container":"audio/mp4","levelCodec":"mp4a.40.2","metadata":{"channelCount":2},"id":"main"}}
audio-stream-controller.ts:153 [log] > [audio-stream-controller]: InitPTS for cc: 0 found from main: 9090/90000
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: WAITING_INIT_PTS->FRAG_LOADING
transmuxer-interface.ts:238 [log] > [transmuxer-interface]: Starting new transmux session for audio sn: 1 track: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: false
        timeOffset: 0
        initSegmentChange: true
91827bb7-6e3c-4997-8e63-86119685b329:10721 [log] > manifest codec:mp4a.40.2, parsed codec:mp4a.40.2, channels:2, rate:44100 (ADTS object type:2 sampling index:4)
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed main sn: 1 of level 1
base-stream-controller.ts:2023 [warn] > [stream-controller]: Found no media in fragment 1 of level 1 resetting transmuxer to fallback to playlist timing
updateLevelTiming @ base-stream-controller.ts:2023
_handleTransmuxerFlush @ base-stream-controller.ts:1037
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warning
base-stream-controller.ts:2093 [log] > [stream-controller]: PARSING->ERROR
main.js:745 Error event: {type: 'mediaError', details: 'fragParsingError', fatal: false, error: Error: Found no media in fragment 1 of level 1 resetting transmuxer to fallback to playlist timing
…, frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:382
trigger @ hls.ts:390
updateLevelTiming @ base-stream-controller.ts:2024
_handleTransmuxerFlush @ base-stream-controller.ts:1037
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warning
base-stream-controller.ts:2093 [log] > [stream-controller]: ERROR->PARSED
base-stream-controller.ts:2040 [log] > [stream-controller]: Parsed main sn: 1 of level 1 (frag:[0.000-2.000])
91827bb7-6e3c-4997-8e63-86119685b329:1076 [log] > Debug logs enabled for "main" in hls.js version 1.6.0
buffer-controller.ts:1010 [warn] > [buffer-controller]: Fragments must have at least one ElementaryStreamType set. type: main level: 1 sn: 1
onFragParsed @ buffer-controller.ts:1010
emit @ index.js:203
emit @ hls.ts:382
trigger @ hls.ts:390
updateLevelTiming @ base-stream-controller.ts:2045
_handleTransmuxerFlush @ base-stream-controller.ts:1037
TransmuxerInterface.onWorkerMessage @ transmuxer-interface.ts:389Understand this warning
buffer-controller.ts:1671 [log] > [buffer-controller]: Blocking operation requested, but no SourceBuffers exist
base-stream-controller.ts:708 [log] > [stream-controller]: Buffered main sn: 1 of level 1 (frag:[0.000-2.000] > buffer:)
base-stream-controller.ts:2093 [log] > [stream-controller]: PARSED->IDLE
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:396 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: FRAG_LOADING->PARSING
audio-stream-controller.ts:962 [log] > [audio-stream-controller]: Init audio buffer, container:audio/mp4, codecs[level/parsed]=[undefined/mp4a.40.2]
buffer-controller.ts:590 [log] > [buffer-controller]: BUFFER_CODECS: "audio" (current SB count 0)
buffer-controller.ts:1337 [log] > [buffer-controller]: checkPendingTracks (pending: 1 codec events expected: 2) {"audio":{"listeners":[],"codec":"mp4a.40.2","container":"audio/mp4","levelCodec":"mp4a.40.2","metadata":{"channelCount":2},"id":"main"}}
transmuxer-interface.ts:396 [log] > [transmuxer.ts]: Flushed audio sn: 1 of track 0
base-stream-controller.ts:2093 [log] > [audio-stream-controller]: PARSING->PARSED
base-stream-controller.ts:2040 [log] > [audio-stream-controller]: Parsed audio sn: 1 of track 0 (frag:[0.000-2.020])
```

### Chrome media internals output

```shell

```

--- a/src/controller/buffer-controller.ts
+++ b/src/controller/buffer-controller.ts
@@ -664,6 +664,18 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
     if (this.sourceBufferCount) {
       return;
     }
+
+    if (
+      this.bufferCodecEventsTotal > 1 &&
+      !this.tracks.video &&
+      !data.video &&
+      data.audio?.id === 'main'
+    ) {
+      // MVP is missing CODECS and only audio was found in main segment (#7524)
+      this.log(`Main audio-only`);
+      this.bufferCodecEventsTotal = 1;
+    }
+
     if (this.mediaSourceOpenOrEnded) {
       this.checkPendingTracks();
     }

